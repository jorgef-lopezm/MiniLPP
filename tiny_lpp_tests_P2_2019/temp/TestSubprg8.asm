global main
extern printf
section .data
true dd 1
false dd 0
x times 5 dd 0
const4 dd 938
const3 dd 253
const2 dd 197
const0 dd 354
const1 dd 642
temp0 dd 0
fmt0 db "%d", 0

section .text
ModificaArreglo:
push ebp
mov ebp, esp
sub esp, 0
mov eax, dword[const0]
mov dword[ebp + 8], eax
mov eax, dword[const1]
mov dword[ebp + 8], eax
mov eax, dword[const2]
mov dword[ebp + 8], eax
mov eax, dword[const3]
mov dword[ebp + 8], eax
mov eax, dword[const4]
mov dword[ebp + 8], eax
mov esp, ebp
pop ebp
ret
main:
push dword[x]
call ModificaArreglo
add esp, 4
mov dword[temp0], eax
push dword[x]
push fmt0
call printf
add esp, 8
push dword[x]
push fmt0
call printf
add esp, 8
push dword[x]
push fmt0
call printf
add esp, 8
push dword[x]
push fmt0
call printf
add esp, 8
push dword[x]
push fmt0
call printf
add esp, 8
ret
