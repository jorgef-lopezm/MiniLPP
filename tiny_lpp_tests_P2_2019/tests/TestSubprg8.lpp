Arreglo [5] de Entero x

Procedimiento ModificaArreglo(var Arreglo [5] de Entero a)
Inicio
    a[1] <- 354
    a[2] <- 642
    a[3] <- 197
    a[4] <- 253
    a[5] <- 938
Fin

Inicio
    LLamar ModificaArreglo(x)
    Escriba x[1]
    Escriba x[2]
    Escriba x[3]
    Escriba x[4]
    Escriba x[5]
Fin


