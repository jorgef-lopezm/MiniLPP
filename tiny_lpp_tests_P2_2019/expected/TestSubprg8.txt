354642197253938